crates/rmf_site_editor/src/mapf_rse/negotiation/mod.rs

```diff
@@ -0,0 +1,357 @@
+                };
+            }
+            Err(err) => match err {
+                NegotiationError::PlanningImpossible(msg) => {
+                    negotiation_task.status = NegotiationTaskStatus::Complete {
+                        elapsed_time,
+                        solution: None,
+                        negotiation_history: Vec::new(),
+                        entity_id_map: HashMap::new(),
+                        error_message: Some(msg),
+                        conflicting_endpoints: Vec::new(),
+                    };
+                }
+                NegotiationError::ConflictingEndpoints(conflicts) => {
+                    negotiation_task.status = NegotiationTaskStatus::Complete {
+                        elapsed_time,
+                        solution: None,
+                        negotiation_history: Vec::new(),
+                        entity_id_map: HashMap::new(),
+                        error_message: None,
+                        conflicting_endpoints: conflicts
+                            .into_iter()
+                            .map(|(a, b)| (bits_string_to_entity(&a), bits_string_to_entity(&b)))
+                            .collect(),
+                    };
+                }
+                NegotiationError::PlanningFailed((negotiation_history, name_map)) => {
+                    negotiation_task.status = NegotiationTaskStatus::Complete {
+                        elapsed_time,
+                        solution: None,
+                        negotiation_history,
+                        entity_id_map: name_map
+                            .into_iter()
+                            .map(|(id, bits_string)| (id, bits_string_to_entity(&bits_string)))
+                            .collect(),
+                        error_message: None,
+                        conflicting_endpoints: Vec::new(),
+                    };
+                }
+            },
+        };
+    }
+}
+
+pub fn start_compute_negotiation(
+    mut commands: Commands,
+    locations: Query<(Entity, &NameInSite, &Point<Entity>), With<LocationTags>>,
+    anchors: Query<(Entity, &Anchor, &Transform)>,
+    negotiation_request: EventReader<NegotiationRequest>,
+    negotiation_params: Res<NegotiationParams>,
+    mut negotiation_debug_data: ResMut<NegotiationDebugData>,
+    mut compute_negotiation_task: Query<(Entity, &mut NegotiationTask)>,
+    current_level: Res<CurrentLevel>,
+    grids: Query<(Entity, &Grid)>,
+    child_of: Query<&ChildOf>,
+    robots: Query<(Entity, &NameInSite, &Pose, &Affiliation<Entity>), With<Robot>>,
+    robot_descriptions: Query<(&DifferentialDrive, &CircleCollision)>,
+    tasks: Query<(&RobotTask, &GoToPlace)>,
+) {
+    if negotiation_request.len() == 0 {
+        return;
+    }
+    if !compute_negotiation_task.is_empty() {
+        let (mut entity, mut negotiation_task) = compute_negotiation_task
+            .single_mut()
+            .expect("Error: Found multiple negotiation tasks!");
```

Comment:
I'd recommend going through all the `expect`s in this PR and making sure they are _really_ critical and required.
An `expect` is pretty much an `unwrap` which means the application will panic if, for example in this case, there is more than one negotiation task.
We should try to aggressively reduce possible panics unless they are really critical and instead have more graceful error handling, most likely reporting to the user through `warn!` or `error!` macros.